.github/workflows/push-registrar-image.yml

```
@@ -0,0 +1,50 @@
+name: Build and Push Docker Images
+
+on:
+  workflow_dispatch:
+
+  schedule:
+    - cron: "0 4 * * *"
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout current repo
+        uses: actions/checkout@v4
+
+      - name: Get tag name
+        id: get-tag-name
+        uses: actions/github-script@v5
+        with:
+          script: |
+            const branchName = context.ref.split('/').slice(-1)[0];
+            const tagName = branchName === 'master' ? 'latest' : branchName;
+            console.log('Will use tag: ' + tagName);
+            return tagName;
+          result-encoding: string
+
+      - name: Build and push Dev Docker image
+        uses: docker/build-push-action@v1
+        with:
+          dockerfile: dockerfiles/registrar.Dockerfile
+          push: true
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+          target: dev
+          repository: edxops/registrar-devstack-dev
```

As the `dev` environment is solely used within the devstack setup, there is no need to add a redundant identifier. Simply specifying `registrar-dev` will suffice. Since we will be updating the image, there's no need to distinguish it from previous versions.
```suggestion
          repository: edxops/registrar-dev
```